pkg/cgroup/cgroup_pids.go

```diff
@@ -0,0 +1,76 @@
+package cgroup
+
+import (
+	"errors"
+	"math"
+	"os"
+	"strings"
+
+	"golang.org/x/sys/unix"
+)
+
+func statPidsV2(dirPath string, stats *Stats) error {
+	current, err := getCgroupParamUint(dirPath, "pids.current")
+	if err != nil {
+		if os.IsNotExist(err) {
+			return statPidsFromCgroupProcs(dirPath, stats)
+		}
+		return err
+	}
+
+	maxVal, err := getCgroupParamUint(dirPath, "pids.max")
+	if err != nil {
+		return err
+	}
+	// If no limit is set, read from pids.max returns "max", which is
+	// converted to MaxUint64 by GetCgroupParamUint. Historically, we
+	// represent "no limit" for pids as 0, thus this conversion.
+	if maxVal == math.MaxUint64 {
+		maxVal = 0
+	}
+
+	stats.PidsStats.Current = current
+	stats.PidsStats.Limit = maxVal
+	return nil
+}
+
+func statPidsFromCgroupProcs(dirPath string, stats *Stats) error {
+	// if the controller is not enabled, let's read PIDS from cgroups.procs
+	// (or threads if cgroup.threads is enabled)
+	contents, err := readCgroupFile(dirPath, "cgroup.procs")
+	if errors.Is(err, unix.ENOTSUP) {
+		contents, err = readCgroupFile(dirPath, "cgroup.threads")
```

Comment:
Is this right? `cgroup.threads` contains the thread IDs not the PIDs. So if we just count the newlines, the number will not be right.

Reply:
Yep. process and thread maps 1 to 1 to linux "task", counting both ways will result in the same pids count.
Here is a fix why reading from cgroup.procs could fail https://github.com/opencontainers/runc/issues/3821

Reply:
That is true, but according to the docs, `cgroup.procs` returns only processes and `cgroup.threads` will return all threads. Those are two different numbers. From what I can tell `cgroup.procs` will never give us the current PID. Also fun fact from the docs
```
       The cgroup.threads file is present in each cgroup (including
       domain cgroups) and can be read in order to discover the set of
       threads that is present in the cgroup.  The set of thread IDs
       obtained when reading this file is not guaranteed to be ordered
       or free of duplicates.
```

So `cgroup.threads` might also return wrong data 😅